lsp/server/src/utils/orgUtils.ts

```diff
@@ -274,17 +242,20 @@ export class OrgUtils {
                         return resolve(objectInfo);
                     }
                 } catch (e) {
-                    return resolve(undefined);
+                    console.log(
+                        `Failed to load entity list from server with error: ${e}`
+                    );
                 }
+                return resolve(undefined);
```

Comment:
Could this resolve go back inside the catch block?

Reply:
If 'connection' is 'undefined', it will never resolve if 'resolve' is put into 'catch'

Reply:
This `connection` is not `undefined` then you should early exit after 
```
const connection = await OrgUtils.getConnection();
if (connection === undefined) {
  return resolve(objectInfo);
}
```
You don't return based on thing that you do in a `try` block.

Without you having telling me I would not have known.

Besides, 
```
return resolve(undefined);
```
was previously in the `catch` block.

Reply:
The 'resolve' is moved out of 'catch' so that it handle two conditions:

1.  Connection is undefined.
2. any error is thrown.
We can put one `resolve(undefined)` inside 'catch`, another check under 'const connection = await OrgUtils.getConnection();'. It is the same thing. It is not a late exit, if connection is undefined, it will go to line 249 immediately.

Reply:
Consider this. Simplified try catch. No post processing of what had happened inside the try block.

```
try {
    const connection = await OrgUtils.getConnection();

    if (connection === undefined || !OrgUtils.entities.includes(objectApiName)) {
        return resolve(undefined);
    }

    const objectInfo = await connection.request(
        `${connection.baseUrl()}/ui-api/object-info/${objectApiName}`
    ) as ObjectInfoRepresentation;

    if (objectInfo !== undefined) {
        this.objectInfoResponseCallback(objectApiName, objectInfo);
    }

    return resolve(objectInfo);

} catch (error) {
    console.log(`Error when calling getObjectInfo: ${error}`);
    return resolve(undefined); // Return undefined in case of an error
}

```

Reply:
Good suggestion!

Reply:
fixed.